### PR TITLE
feat: add thanos article

### PR DESCRIPTION
**Pull request by ArchiFleKs**

### CODE REVIEW

File: content/blog/thanos-monitoring.en.md [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+the number of cluster from which you want to get metrics.
+
+##### Multiple Grafana
+
+In this setup each cluster has its own Grafana with its own set of dashboards
+which can be a pain to maintain.
+
+##### Storing metrics is expensive
+
+Prometheus stores metrics on disk, you have to make a choices between storage
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: `choice`

File: content/blog/thanos-monitoring.en.md [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+example. The drawback of this solution is that you cannot make calculation based
+on different data sources.
+
+##### Prometheus Federation
+
+[Prometheus
+federation](https://prometheus.io/docs/prometheus/latest/federation/) allow
+scraping prometheuses from prometheus, this solution works well when you are not
+scraping a lot of metrics. At scale if scraping all your other prometheuses
+takes longer than the scrape interval this is going to cause issue.
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: What do you think of: `At scale, if the scrape duration of all your prometheus targets takes longer than the scrape interval, you might encounter some severe issues.` ?

File: content/blog/thanos-monitoring.en.md [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+
+##### Prometheus remote write
+
+While [remote writing](https://prometheus.io/docs/practices/remote_write/) is a
+solution (and is also implemented by Thanos receiver), we will not discuss the
+"pushing metrics" part in this article, you can read about the pros and cons of
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: `article, you` -> `article. You` (?)

File: content/blog/thanos-monitoring.en.md [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+It is recommended to push metrics as a last resort or when not trusting multiple
+cluster or tenant (for example when building a prometheus as a service
+offering). Anyway this might be a topic for a further article but we will focus
+on the scrapping here.
+
+### Enter Thanos
+
+![](/images/logos/thanos.svg#logosize)
+
+[Thanos](https://thanos.io/) is a "*Open source, highly available Prometheus
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: `a` -> `an` ?

File: content/blog/thanos-monitoring.en.md [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+      observee clusters
+
+Thanos component deployed:
+
+  * All thanos component are deployed
+  * Query Frontend which serve as a datasource endpoints for Grafana
+  * Storegateway are deployed to query the observer bucket
+  * Query will perform query to the storegateways and the other querier
+
+Additionnals Thanos components deployed:
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: `Additionnals` -> `Additionnal` (?)

File: content/blog/thanos-monitoring.en.md [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+[Thanos store](https://thanos.io/tip/components/store.md/) acts a gateway to
+translate query to remote object storage. It can also cache some information on
+local storage. Basically this is the component that allows you to query an
+object store for metrics. This component acts as a store for Thanos Query
+
+##### Thanos Compactor
+
+[Thanos compactor](https://thanos.io/tip/components/compact.md/) is a singleton
+(it is not scalable) that is responsible for compacting and downsampling the
+metrics stored inside an object store. Downsampling is the fact of loosing
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: `the fact` -> `the action` ?

File: content/blog/thanos-monitoring.en.md [hunk 7]
```diff
@@ -0,0 +1,524 @@
...
+granularity on your metrics over time: for example you may want to keep your
```
Comment (tbobm) [inline content/blog/thanos-monitoring.en.md]: `time: for example you may`

maybe one of:
- `time. You may`
- `time. For example you may`

**Review by tbobm — CHANGES_REQUESTED** [review]
Really nice article! :ok_hand:

File: content/blog/thanos-monitoring.en.md [hunk 8]
```diff
@@ -0,0 +1,525 @@
+---
+Title: Multi Cluster Monitoring with Thanos
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: clusters ?

File: content/blog/thanos-monitoring.en.md [hunk 9]
```diff
@@ -0,0 +1,525 @@
...
+Date: 2021-02-23T11:00:00+00:00
+Category: Kubernetes
+Summary: Monitor multiple Kubernetes Cluster with Prometheus and Thanos
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: Clusters ?

File: content/blog/thanos-monitoring.en.md [hunk 10]
```diff
@@ -0,0 +1,525 @@
...
+[Thanos sidecar](https://thanos.io/tip/components/sidecar.md/) is available out
+of the box with [Prometheus
+Operator](https://github.com/prometheus-operator/prometheus-operator) and [Kube
+Prometheus
+Stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)
+and can be deploy easily. This component act as a store for Thanos Query.
+
+##### Thanos Store
+
+[Thanos store](https://thanos.io/tip/components/store.md/) acts a gateway to
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: acts as*

File: content/blog/thanos-monitoring.en.md [hunk 11]
```diff
@@ -0,0 +1,525 @@
...
+Frontend](https://thanos.io/tip/components/query-frontend.md/) acts a frontend
+for Thanos Query, its goal is to split large query into multiple smaller queries
+and also to cache the query result (either in memory or in a memcached)
+
+There are also other components such as Thanos Receive in the case of remote
+write but this is still not the topic of this article.
+
+### Multi Cluster Architecture
+
+There are multiple way to deploy these component into multiples Kubernetes
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: components

Comment (rguichard) [reply content/blog/thanos-monitoring.en.md]: multiple

File: content/blog/thanos-monitoring.en.md [hunk 12]
```diff
@@ -0,0 +1,525 @@
...
+
+A CA is generated for the observer cluster:
+
+  * this CA will be trusted by the observee clusters ingress sidecar
+  * TLS certs are generated for Thanos querier components that will query the
+      observee clusters
+
+Thanos component deployed:
+
+  * All thanos component are deployed
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: components

File: content/blog/thanos-monitoring.en.md [hunk 13]
```diff
@@ -0,0 +1,525 @@
...
+  * Query Frontend which serve as a datasource endpoints for Grafana
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: endpoint

File: content/blog/thanos-monitoring.en.md [hunk 14]
```diff
@@ -0,0 +1,525 @@
...
+      default_global_requests = true
+      default_global_limits   = false
+      bucket                  = "thanos-store-pio-thanos-observee"
+      region                  = "eu-west-3"
+    }
+  ```
+
+#### Observee cluster
+
+Observee cluster are Kubernetes cluster with minimal prometheus/thanos
```
Comment (rguichard) [inline content/blog/thanos-monitoring.en.md]: clusters

**Review by rguichard — COMMENTED** [review]
Very good ! I missed a lot of prometheus/Prometheus - thanos/Thanos. I think we should use an uppercase first letter everywhere.

There are two PNG bigger than 1MB. Can you make them smaller ?

I'll make the OG image

**Review by rguichard — APPROVED** [review]
lgtm

**Pull request merged**